lib/pushgateway.js

```diff
@@ -5,7 +5,11 @@ var http = require('http');
 var https = require('https');
 var register = require('./register');
 
-function Pushgateway(gatewayUrl, options) {
+function Pushgateway(gatewayUrl, options, registry) {
+	if(!registry) {
+		registry = register;
```

Comment:
Same here, don't mutate input variables to the function - will bite us in the end